tests/inc/test-class-myyoast-api-request.php

```diff
@@ -0,0 +1,226 @@
+class WPSEO_MyYoast_Api_Request_Test extends WPSEO_UnitTestCase {
+
+	/**
+	 * Tests firing the request (happy path).
+	 *
+	 * @covers WPSEO_MyYoast_Api_Request::fire
+	 * @covers WPSEO_MyYoast_Api_Request::get_response
+	 */
+	public function test_fire() {
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Api_Request' )
+			->setMethods( array( 'do_request', 'decode_response' ) )
+			->setConstructorArgs( array( 'endpoint' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'do_request' )
+			->will( $this->returnValue( 'raw_response' ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'decode_response' )
+			->will( $this->returnValue( 'response' ) );
+
+		$this->assertTrue( $instance->fire() );
+		$this->assertEquals( 'response', $instance->get_response() );
+	}
+
+	/**
+	 * Tests firing the request with an exception being thrown.
+	 *
+	 * @covers WPSEO_MyYoast_Api_Request::fire
+	 * @covers WPSEO_MyYoast_Api_Request::get_error_message
+	 */
+	public function test_fire_with_bad_request_exception_being_thrown() {
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Api_Request' )
+			->setMethods( array( 'do_request', 'decode_response' ) )
+			->setConstructorArgs( array( 'endpoint' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'do_request' )
+			->will( $this->throwException( new WPSEO_MyYoast_Bad_Request_Exception( 'Request went wrong' ) ) );
+
+		$instance
+			->expects( $this->never() )
+			->method( 'decode_response' );
+
+		$this->assertFalse( $instance->fire() );
+		$this->assertAttributeEquals( 'Request went wrong', 'error_message', $instance );
+		$this->assertEquals( 'Request went wrong', $instance->get_error_message() );
+	}
+
+	/**
+	 * Tests firing the request with an exception being thrown.
+	 *
+	 * @covers WPSEO_MyYoast_Api_Request::fire
+	 */
+	public function test_fire_with_authentication_exception_being_thrown() {
+		if ( ! WPSEO_Utils::has_access_token_support() ) {
+			$this->markTestSkipped( 'Depends on PHP 5.6 functionality.' );
+		}
+
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Api_Request' )
+			->setMethods( array( 'do_request', 'decode_response', 'get_access_token' ) )
+			->setConstructorArgs( array( 'endpoint' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->exactly( 2 ) )
+			->method( 'do_request' )
+			->will(
+				$this->onConsecutiveCalls(
+				$this->throwException( new WPSEO_MyYoast_Authentication_Exception( 'Authentication failed' ) ),
+					'response'
+				)
+			);
+
+		$instance
+			->expects( $this->once() )
+			->method( 'decode_response' );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'get_access_token' );
+
+		$this->assertTrue( $instance->fire() );
+	}
+
+	/**
+	 * Tests firing the request with an exception being thrown.
+	 *
+	 * @covers WPSEO_MyYoast_Api_Request::fire
+	 */
+	public function test_fire_with_retrieving_new_access_token_result_in_authentication_error() {
```

Comment:
I don't see a test with retrieving an access token, followed by a successful request

Reply:
It turned out I overlooked that test 😄